### PR TITLE
etcdutil: factor out wait ready

### PR DESCRIPTION
**Pull request by xiang90**
/cc @diegs 

we want to reuse the cluster wait logic for recovery.

### CODE REVIEW

File: pkg/util/etcdutil/migrate.go [hunk 1]
```diff
@@ -102,34 +101,22 @@ func waitEtcdTPRReady(restClient restclient.Interface, ns string) error {
...
+	svcb, err := ioutil.ReadFile(svcPath)
 	if err != nil {
 		return err
 	}
-	if err := kubecli.CoreV1().RESTClient().Post().RequestURI(fmt.Sprintf("/api/v1/namespaces/%s/services", api.NamespaceSystem)).SetHeader("Content-Type", "application/json").Body(svc).Do().Error(); err != nil {
+	if err := kubecli.CoreV1().RESTClient().Post().RequestURI(fmt.Sprintf("/api/v1/namespaces/%s/services", api.NamespaceSystem)).SetHeader("Content-Type", "application/json").Body(svcb).Do().Error(); err != nil {
+		return err
+	}
+
+	svc, err := kubecli.CoreV1().Services(api.NamespaceSystem).Get(bootstrapEtcdServiceName, v1.GetOptions{})
```
Comment (xiang90) [inline pkg/util/etcdutil/migrate.go]: @diegs 

is there a reason we also put this get func into the wait loop? if 108 returns, we should be able to get the service, right?

Comment (diegs) [inline pkg/util/etcdutil/migrate.go]: Doesn't really matter either way, you can pull it out if you like.

File: pkg/util/etcdutil/util.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// WaitReady waits the etcd server ready to serve client requests.
+func WaitReady(endpoint string) error {
```
Comment (diegs) [inline pkg/util/etcdutil/util.go]: `waitClusterReady`? As in, `etcdutil.WaitClusterReady(endpoint)`

**Comment by xiang90** [comment]
ignore e2e-selfetcd  according to @pbx0.

merging. thanks for the review @diegs !

**Pull request merged**